branches.go

```diff
@@ -183,6 +185,18 @@ func cleanCommitBranch(dir string, preservedPaths []string) error {
 	return err
 }
 
+// copyBranchContents copies the entire contents of the source directory to the
+// destination directory, except for .git.
+func copyBranchContents(srcDir, dstDir string) error {
+	// nolint: gosec
+	if _, err := libExec.Exec(
+		exec.Command("cp", "-r", srcDir, dstDir),
```

Comment:
`cp` may not exist in trimmed-down or distroless-like images, which is an eventual direction we need to consider for kargo or other utilities built on kargo-render. Do you think we could replace this with a golang implementation? e.g.:

https://stackoverflow.com/questions/51779243/copy-a-folder-in-go

Reply:
I'm open to that, but would note two things:

We use `cp` in other places besides this (simply because the Go stdlib doesn't have a good alternative -- which would be why those libs all exist).

I love distroless and would much prefer that we we _were_ distroless, but I believe the fact that we exec out to the `git`CLI (for reasons I believe you're familiar with) present a much larger hurdle to achieving distroless than `cp` does.

Is this a bridge you'd be open to crossing at a later time?

Reply:
Yep, if we are already using `cp` else where, then there's no sense fixing it here and we can address it as separate feature.